panel/src/components/Forms/Writer/Writer.vue

```diff
@@ -252,12 +252,15 @@ export default {
 			const plugins = window.panel.plugins.writerMarks ?? {};
 			const marks = {};
 
-			// take each extenstion object and turn
+			// take each extension object and turn
 			// it into an instance that extends the Mark class
 			for (const name in plugins) {
+				const plugin = plugins[name];
+				plugin.name ??= name;
+
 				marks[name] = Object.create(
 					Mark.prototype,
-					Object.getOwnPropertyDescriptors(plugins[name])
+					Object.getOwnPropertyDescriptors(plugin)
```

Comment:
This is actually exactly what I had in mind, just didn't realize this is already the right spot when I wrote from my phone.

Might have written is just as 

```suggestion
					Object.getOwnPropertyDescriptors({ name, ...plugins[name] })
```

to make it a one-line

Reply:
That's what I need 💯